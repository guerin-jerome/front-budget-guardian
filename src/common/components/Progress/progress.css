.progress-bar {
  height: 10px;
  background-color: #eee;
}

.progress {
  height: 100%;
  background-color: #0077c2;
}

.progress.error {
  background-color: #b33939;
}
