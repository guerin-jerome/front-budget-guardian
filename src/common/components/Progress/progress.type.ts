export type ProgressProps = {
  width: string;
  hasError: boolean;
};
