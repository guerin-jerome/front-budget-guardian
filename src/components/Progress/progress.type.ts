export type TProgress = {
  width: string;
};
