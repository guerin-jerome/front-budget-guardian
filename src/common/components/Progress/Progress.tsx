import { ProgressProps } from "./progress.type";
import classNames from "classnames";
import "./progress.css";

export const Progress = ({ width, hasError }: ProgressProps) => (
  <div className="progress-bar">
    <div
      className={classNames("progress", { error: hasError })}
      style={{ width }}
    />
  </div>
);
