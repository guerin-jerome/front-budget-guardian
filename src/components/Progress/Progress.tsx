import { TProgress } from "./progress.type";
import "./progress.css";

export const Progress = ({ width }: TProgress) => (
  <div className="progress-bar">
    <div className="progress" style={{ width }} />
  </div>
);
