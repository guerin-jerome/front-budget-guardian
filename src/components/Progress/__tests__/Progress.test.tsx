import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Progress } from "../Progress";

describe("<Progress />", () => {
  it("matchs snapshot", () => {
    const { container } = render(<Progress width="10%" />);
    expect(container).toMatchSnapshot();
  });
});
