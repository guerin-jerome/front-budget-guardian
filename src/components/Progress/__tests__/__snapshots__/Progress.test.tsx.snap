// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Progress /> > matchs snapshot 1`] = `
<div>
  <div
    class="progress-bar"
  >
    <div
      class="progress"
      style="width: 10%;"
    />
  </div>
</div>
`;
